drivers/video/fbdev/rm68090fb.c

```diff
@@ -0,0 +1,1124 @@
+  lcdc_wr_dat(0x0001);
+  lcdc_wr_cmd(0x35); 
+  lcdc_wr_dat(0x0204);
+  lcdc_wr_cmd(0x36); 
+  lcdc_wr_dat(0x0004);
+  lcdc_wr_cmd(0x37); 
+  lcdc_wr_dat(0x0407);
+  lcdc_wr_cmd(0x38); 
+  lcdc_wr_dat(0x0000);
+  lcdc_wr_cmd(0x39); 
+  lcdc_wr_dat(0x0404);
+  lcdc_wr_cmd(0x3C); 
+  lcdc_wr_dat(0x0402);
+  lcdc_wr_cmd(0x3D); 
+  lcdc_wr_dat(0x0004);
+  //---------------  RAM Address Control ----------------//
+  lcdc_wr_cmd(0x50); 
+  lcdc_wr_dat(0x0000); // Set GRAM Horizontal Start Address
+  lcdc_wr_cmd(0x51); 
+  lcdc_wr_dat(0x00EF); // Set GRAM Horizontal End Address
+  lcdc_wr_cmd(0x52); 
+  lcdc_wr_dat(0x0000); // Set GRAM Vertical Start Address
+  lcdc_wr_cmd(0x53); 
+  lcdc_wr_dat(0x013F); // Set GRAM Vertical End Address
+  //---------------  Panel Image Control -----------------//
+  lcdc_wr_cmd(0x60); 
+  lcdc_wr_dat(0x2700); // Set Gate Scan line
+  lcdc_wr_cmd(0x61); 
+  lcdc_wr_dat(0x0001); // Set NDL); VLE); REV
+  lcdc_wr_cmd(0x6A); 
+  lcdc_wr_dat(0x0000); // Set Scrolling line
+  //---------------  Panel Interface Control---------------//
+  lcdc_wr_cmd(0x90); 
+  lcdc_wr_dat(0x0010);
+  lcdc_wr_cmd(0x92); 
+  lcdc_wr_dat(0x0000);
+  //--------------- Display On-------------------------------//
+  lcdc_wr_cmd(0x07); 
+  lcdc_wr_dat(0x0133); // Display on
+  lcdc_wr_cmd(0x22);
+
+
+  writel(0xffffffff, iomm.gpio + PD_DATA);
+  ret = readl(iomm.gpio + PD_CFG0);
+  ret&= 0x0000000f;
+  ret|= 0x22222220;
+  writel(ret, iomm.gpio + PD_CFG0);
+  writel(0x22222202, iomm.gpio + PD_CFG1);
+  writel(0x00222222, iomm.gpio + PD_CFG2);
+
+}
+
+static void suniv_enable_irq(struct myfb_par *par)
+{
+  int ret;
+
+  par->have_te = 0;
+  par->lcdc_irq = platform_get_irq(par->pdev, 0);
+  if (par->lcdc_irq < 0) {
+    printk("%s, failed to get irq number for lcdc irq\n", __func__);
+  }
+  else{
+	  ret = request_irq(par->lcdc_irq, lcdc_irq_handler, IRQF_SHARED, "miyoo_lcdc_irq", par);
+	  if(ret){
+		  printk("%s, failed to register lcdc interrupt(%d)\n", __func__, par->lcdc_irq);
+	  }
+  }
+  
+  par->gpio_irq = -1;//gpio_to_irq(SLCD_TE);
+  if (par->gpio_irq < 0) {
+    //printk("%s, failed to get irq number for gpio irq\n", __func__);
+  }
+  else{
+	  ret = request_irq(par->gpio_irq, gpio_irq_handler, IRQF_TRIGGER_RISING, "miyoo_gpio_irq", par);
+	  if(ret){
+		  printk("%s, failed to register gpio interrupt(%d)\n", __func__, par->gpio_irq);
+	  }
+  }
+}
+
+static void suniv_lcdc_init(struct myfb_par *par)
+{
+  uint32_t ret=0, bp=0, total=0, ver=0;
+	uint32_t h_front_porch = 8;
+	uint32_t h_back_porch = 8;
+	uint32_t h_sync_len = 1;
+	uint32_t v_front_porch = 8;
+	uint32_t v_back_porch = 8;
+	uint32_t v_sync_len = 1;
+      
+  ser_init();
+  ser_wr_cmd(0x04);
+  if(ser_rd_dat() == 0x42c2a97f){
+    ver = 4;
+  }
+  ser_deinit();
+
+  if (version == 4 && ver != 4) {
+    printk("Warning: LCD controller not detected as version 4, but being overridden by module_param as version 4 (detection 1 of 2)");
+    ver = version;
+  }
```

Comment:
ser_init();
  ser_wr_cmd(0x04);
  if(ser_rd_dat() == 0x42c2a97f){
    ver = 4;
  }
  ser_deinit();

  if (version == 4 && ver != 4) {
    printk("Warning: LCD controller not detected as version 4, but being overridden by module_param as version 4 (detection 1 of 2)");
    ver = version;
  }

to remove